test/invariant/fuzz/handlers/HandlerAllo.t.sol

```diff
@@ -33,65 +34,299 @@ contract HandlerAllo is Setup {
             )
         );
 
-        if (succ) ghost_poolIds.push(abi.decode(ret, (uint256)));
+        if (succ) {
+            uint256 _poolId = abi.decode(ret, (uint256));
+            ghost_poolIds.push(_poolId);
+            ghost_poolAdmins[_poolId] = msg.sender;
+        }
     }
 
-    function handler_updatePoolMetadata(
-        uint256 _idSeed,
-        uint256 _metadataProtocol,
-        string calldata _data
-    ) public {
+    function handler_updatePoolMetadata(uint256 _idSeed, uint256 _metadataProtocol, string calldata _data) public {
         // Needs at least one pool
         if (ghost_poolIds.length == 0) return;
 
         _idSeed = _idSeed % ghost_poolIds.length;
         uint256 poolId = ghost_poolIds[_idSeed];
 
         // Get the profile ID
-        IRegistry.Profile memory profile = registry.getProfileByAnchor(
-            _ghost_anchorOf[msg.sender]
-        );
+        IRegistry.Profile memory profile = registry.getProfileByAnchor(_ghost_anchorOf[msg.sender]);
 
         // Avoid EOA
         if (profile.anchor == address(0)) return;
 
-        Metadata memory metadata = Metadata({
-            protocol: _metadataProtocol,
-            pointer: _data
-        });
+        Metadata memory metadata = Metadata({protocol: _metadataProtocol, pointer: _data});
 
         // Update the pool metadata - will revert on wrong anchor
-        targetCall(
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updatePoolMetadata.selector, poolId, metadata));
+    }
+
+    function handler_updatePercentFee(uint256 _newPercentFee) public {
+        _newPercentFee = bound(_newPercentFee, 0, 1e18);
+
+        // Update the percent fee - will revert if caller is not the owner
+        targetCall(address(allo), 0, abi.encodeWithSelector(IAllo.updatePercentFee.selector, _newPercentFee));
```

Comment:
nit (and I should fix it in my branches too): we should *really* use abi.encodeCall instead (will silently fail if we mess up arg type/param ordering/etc)